llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -1620,3 +1620,15 @@ bool RISCVTTIImpl::isLSRCostLess(const TargetTransformInfo::LSRCost &C1,
                   C2.NumIVMuls, C2.NumBaseAdds,
                   C2.ScaleCost, C2.ImmCost, C2.SetupCost);
 }
+
+bool RISCVTTIImpl::isLegalMaskedCompressStore(Type *DataTy, Align Alignment) {
+  auto *VTy = dyn_cast<VectorType>(DataTy);
+  if (!VTy || VTy->isScalableTy() || !ST->hasVInstructions())
+    return false;
+
+  if (!TLI->isLegalElementTypeForRVV(
+          TLI->getValueType(DL, VTy->getElementType())))
+    return false;
+
+  return getRegUsageForType(VTy) <= 8;
```

Comment:
Why is this needed? The backend should be able to split a compressed store.

Reply:
It's not for the store, but for the compress. Splitting it for LMUL >= 16 is doable, but is not simple

Reply:
The splitting will happen while it is still a compressedstore. The compress instruction isn't created until after the split is already done.